### PR TITLE
Add Optional CUP Variants component

### PR DESCRIPTION
**Pull request by TyroneMF**
- currently only has a lonesome C130J in Theseus colours.

### CODE REVIEW

File: optionals/variants_cup/CfgVehicles.hpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        displayName = CSTRING(C130J_Theseus_Name);
+        editorPreview = QPATHTOF(UI\C130J_B_Theseus.jpg);
+        crew = QCLASS(Unit_I_PilotHeli);
+        typicalCargo[] = {QCLASS(Unit_I_Engineer)};
+        hiddenSelections[] = {"camo1","camo2"};
+        hiddenSelectionsTextures[] = {
+            QPATHTOF(data\theseus_c130j_body_co.paa),
+            QPATHTOF(data\theseus_c130j_wings_co.paa)
+        };
+        class textureSources {
```
Comment (jonpas) [inline optionals/variants_cup/CfgVehicles.hpp:21]:
```suggestion
        class TextureSources {
```

File: optionals/variants_cup/CfgVehicles.hpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            class TACS {
```
Comment (jonpas) [inline optionals/variants_cup/CfgVehicles.hpp:22]:
```suggestion
            class ADDON {
```

**Review by jonpas — CHANGES_REQUESTED** [review]

File: optionals/variants_cup/CfgVehicles.hpp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        scope = 2;
+        forceInGarage = 1;
+        side = 2;
+        faction = QCLASS(IND);
+        author = CSTRING(C130J_Author);
+        displayName = CSTRING(C130J_Theseus_Name);
+        editorPreview = QPATHTOF(UI\C130J_B_Theseus.jpg);
+        crew = QCLASS(Unit_I_PilotHeli);
+        typicalCargo[] = {QCLASS(Unit_I_Engineer)};
+        hiddenSelections[] = {"camo1","camo2"};
```
Comment (jonpas) [inline optionals/variants_cup/CfgVehicles.hpp:16]:
```suggestion
        hiddenSelections[] = {"camo1", "camo2"};
```

**Review by jonpas — CHANGES_REQUESTED** [review]

File: optionals/variants_cup/config.cpp [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+class CfgPatches {
+    class ADDON {
+        name = COMPONENT_NAME;
+        units[] = {};
+        weapons[] = {};
+        requiredVersion = REQUIRED_VERSION;
+        requiredAddons[] = {"tacs_units", "CUP_AirVehicles_C130J"};
+        author = ECSTRING(main,Author);
+        authors[] = {"CUP Team", "GilleeDoo", "Tyrone"};
```
Comment (jonpas) [inline optionals/variants_cup/config.cpp:11]:
```suggestion
        authors[] = {"GilleeDoo", "Tyrone"};
```
We don't put it in Arcadian either. It is mentioned in the `README.md` - stick to "authors of what was added in this mod".

File: optionals/variants_cup/CfgVehicles.hpp [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        };
+        class TextureSources {
+            class ADDON {
+                displayname = CSTRING(C130J_Theseus_Texture_Name);
+                author = CSTRING(C130J_Author);
+                textures[] = {
+                    QPATHTOF(data\theseus_c130j_body_co.paa),
+                    QPATHTOF(data\theseus_c130j_wings_co.paa)
+                };
+                factions[] = {};
```
Comment (jonpas) [inline optionals/variants_cup/CfgVehicles.hpp:29]: Why empty?

File: optionals/variants_cup/CfgVehicles.hpp [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    class CLASS(C130J_I_Theseus): CUP_B_C130J_USMC {
+        MACRO_AIRVEHICLE_CARGO
+        dlc = QUOTE(PREFIX);
+        scope = 2;
+        forceInGarage = 1;
+        side = 2;
+        faction = QCLASS(IND);
+        author = CSTRING(C130J_Author);
+        displayName = CSTRING(C130J_Theseus_Name);
+        editorPreview = QPATHTOF(UI\C130J_B_Theseus.jpg);
```
Comment (jonpas) [inline optionals/variants_cup/CfgVehicles.hpp:13]:
```suggestion
        editorPreview = QPATHTOF(ui\C130J_B_Theseus.jpg);
```

**Review by jonpas — CHANGES_REQUESTED** [review]
Background of UI preview image looks wrong.

`UI` should be lower-cased `ui`.

File: optionals/variants_cup/CfgVehicles.hpp (hunk 5 above)
Comment (TyroneMF) [reply optionals/variants_cup/CfgVehicles.hpp:29]: Fixed in next commit.

**Review by jonpas — COMMENTED** [review]
Need to take a look at preview image color grading.

**Comment by TyroneMF** [comment]
Can't replicate the BI color grading, always comes out the same for me.

File: optionals/variants_cup/config.cpp [hunk 7]
```diff
@@ -0,0 +1,17 @@
+#include "script_component.hpp"
+
+class CfgPatches {
+    class ADDON {
+        name = COMPONENT_NAME;
+        units[] = {};
```
Comment (jonpas) [inline optionals/variants_cup/config.cpp:6]: Shouldn't the vehicle be here for Zeus?

**Review by jonpas — APPROVED** [review]

**Pull request merged**